File: backend/app/schemas/stock.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class MovingAveragesData(BaseModel):
    ma_50: Optional[float] = None
    ma_100: Optional[float] = None
    ma_150: Optional[float] = None
    ma_200_day: Optional[float] = None
    ma_200_week: Optional[float] = None


class HighLowRange(BaseModel):
    week_52_high: Optional[float] = None
    week_52_low: Optional[float] = None
    current_price: float
    position_percent: Optional[float] = None  # Position in 52-week range (0-100)


class StockData(BaseModel):
    symbol: str
    name: str
    current_price: float
    change_24h: Optional[float] = None
    change_24h_percent: Optional[float] = None
    moving_averages: MovingAveragesData
    high_low_range: HighLowRange
    last_updated: datetime


class PricePoint(BaseModel):
    date: datetime
    price: float


class StockChartData(BaseModel):
    symbol: str
    prices: list[PricePoint]

File: frontend/src/types/stock.ts
export interface MovingAveragesData {
  ma_50: number | null;
  ma_100: number | null;
  ma_150: number | null;
  ma_200_day: number | null;
  ma_200_week: number | null;
}

export interface HighLowRange {
  week_52_high: number | null;
  week_52_low: number | null;
  current_price: number;
  position_percent: number | null;
}

export interface StockData {
  symbol: string;
  name: string;
  current_price: number;
  change_24h: number | null;
  change_24h_percent: number | null;
  moving_averages: MovingAveragesData;
  high_low_range: HighLowRange;
  last_updated: string;
}

export interface PricePoint {
  date: string;
  price: number;
}

export interface StockChartData {
  symbol: string;
  prices: PricePoint[];
}
